x-pack/plugins/apm/public/components/shared/span_links/span_links_table.tsx

```diff
@@ -0,0 +1,234 @@
+          );
+        }
+        return (
+          <EuiFlexGroup alignItems="center" gutterSize="xs" responsive={false}>
+            <EuiFlexItem grow={false}>
+              <EuiIcon type="stopSlash" size="m" color="subdued" />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              {i18n.translate('xpack.apm.spanLinks.table.serviceName.unknown', {
+                defaultMessage: 'Unknown',
+              })}
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        );
+      },
+    },
+    {
+      field: 'spanId',
+      name: i18n.translate('xpack.apm.spanLinks.table.span', {
+        defaultMessage: 'Span',
+      }),
+      sortable: true,
+      render: (_, { spanId, traceId, details }) => {
+        if (details) {
+          return (
+            <EuiFlexGroup
+              alignItems="center"
+              gutterSize="xs"
+              responsive={false}
+            >
+              <EuiFlexItem grow={false}>
+                <EuiIcon
+                  type={getSpanIcon(details.spanType, details.spanSubtype)}
+                  size="l"
+                />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiLink
+                  href={link('/link-to/transaction/{transactionId}', {
+                    path: { transactionId: details.transactionId || spanId },
+                    query: { waterfallItemId: spanId },
+                  })}
+                >
+                  {details.spanName}
+                </EuiLink>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          );
+        }
+        return `${traceId}-${spanId}`;
+      },
+    },
+    {
+      field: 'duration',
+      name: i18n.translate('xpack.apm.spanLinks.table.spanDuration', {
+        defaultMessage: 'Span duration',
+      }),
+      sortable: true,
+      width: '150',
+      render: (_, { details }) => {
+        return (
+          <EuiText size="s" color="subdued">
+            {asDuration(details?.duration)}
+          </EuiText>
+        );
+      },
+    },
+    {
+      field: 'actions',
+      name: 'Actions',
+      width: '100',
+      render: (_, { spanId, traceId, details }) => {
+        const id = `${traceId}:${spanId}`;
+        return (
+          <EuiPopover
+            button={
+              <EuiButtonIcon
+                aria-label="Edit"
+                iconType="boxesHorizontal"
+                onClick={() => {
+                  setIdActionMenuOpen(id);
+                }}
+              />
+            }
+            isOpen={idActionMenuOpen === id}
+            closePopover={() => {
+              setIdActionMenuOpen(undefined);
+            }}
+          >
+            <EuiFlexGroup direction="column" gutterSize="s">
+              {details && (
+                <EuiFlexItem>
+                  <EuiLink
+                    href={link('/link-to/transaction/{transactionId}', {
+                      path: { transactionId: details.transactionId || spanId },
```

Comment:
Hmm, this seems off. If we're linking to a span rather than a transaction we should probably have a separate link for that?

Reply:
yes, you're right. So I'll only render this link when `details.transactionId` is not empty and use it to create the link not the `spanId`.

Reply:
Same for the other link that opens the detailed flyout.